openid4vc-high-assurance-interoperability-profile-1_0.md

```diff
@@ -259,17 +260,22 @@ A Credential Format Profile for Credentials complying with IETF SD-JWT VCs [@!I-
 
 # Crypto Suites
 
-Issuers, holders and verifiers MUST support P-256 (secp256r1) as a key type with ES256 JWT algorithm for signing and signature validation whenever this profiles requires to do so:
+Cryptography is required by the following operations:
+
+- to sign and validate the signature on the Wallet Attestation and its proof of possession
+- to sign and validate the Issuer's signature on the Verifiable Credential
+- to sign and validate the Holder's signature on the Verifiable Presentation
+- to sign and validate the Verifier's signature on the Presentation Request
+
+Issuers, Holders, and Verifiers MUST support P-256 (secp256r1) as a key type with the ES256 JWT algorithm [@!RFC7518] for the creation and the verification of the above signatures.
```

Comment:
```suggestion
Issuers, Holders, and Verifiers MUST at least support P-256 (secp256r1) as a key type with the ES256 JWT algorithm [@!RFC7518] for the creation and the verification of the above signatures.
```

Reply:
- the intent is to allow using non P-256 algs. credential does not have to be signed using P-256. if credentials signed using other algs wants to use HAIP, they need to define
- this intends to say P256 is a fall back and is minimum mandatory to implement
- there might be secure areas that might have to add P256 compliance just to be HAIP compliant? which might be limiting
- these requirements might make sense for issuance but not presentation?

Reply:
- for the key for issuer signed data, Wallet has to accept P256 for SD-JWT/IssuerSigned (issuer communicates what it supports in the issuer metadata, can be P384).for the key for holder binding, issuer **has to accept** P256 for KB JWT/DeviceSigned; 
- for presentation, for RP, mandate P256 to use it to validate. no  requirement for the wallet has the credential, it already has it.

Reply:
How does the text contradict your statements?

Reply:
@Sakurann from my perspective, the MUST and MUST at least requirements produce the same expected result. Implementers must support P256 regardless of whether the requirement is stated as MUST or MUST at least.